projects/components/src/table/table.component.ts

```diff
@@ -840,10 +841,11 @@ export class TableComponent
 
     this.dataSource?.disconnect();
     this.dataSource = this.buildDataSource();
-
-    this.dataSource?.loadingStateChange$.subscribe(() => {
-      this.tableService.updateFilterValues(this.columnConfigsSubject.value, this.dataSource!); // Mutation! Ew!
-    });
+    this.subscriptionLifecycle.add(
+      this.dataSource?.loadingStateChange$.subscribe(() => {
```

Comment:
also finite. The data source's disconnect method completes this (however this is questionable; that means it cannot be connect'ed again).